internal/services/dynatrace/dynatrace_monitor_resource_test.go

```diff
@@ -175,6 +191,86 @@ resource "azurerm_dynatrace_monitor" "test" {
 `, template, data.RandomInteger, r.dynatraceInfo.UserFirstName, r.dynatraceInfo.UserLastName, r.dynatraceInfo.UserEmail, r.dynatraceInfo.UserPhoneNumber, r.dynatraceInfo.UserCountry)
 }
 
+func (r MonitorsResource) complete(data acceptance.TestData) string {
+	template := r.template(data)
+	return fmt.Sprintf(`
+%[1]s
+
+resource "azurerm_dynatrace_monitor" "test" {
+  name                     = "acctestacc%[2]d"
+  resource_group_name      = azurerm_resource_group.test.name
+  location                 = azurerm_resource_group.test.location
+  marketplace_subscription = "Active"
+
+  identity {
+    type = "SystemAssigned"
+  }
+
+  user {
+    first_name   = "%[4]s"
+    last_name    = "%[5]s"
+    email        = "%[6]s"
+    phone_number = "%[7]s"
+    country      = "%[8]s"
+  }
+
+  plan {
+    usage_type    = "COMMITTED"
+    billing_cycle = "MONTHLY"
+    plan          = "azureportalintegration_privatepreview@TIDgmz7xq9ge3py"
+  }
+
+  tags = {
+    environment = "Dev"
+  }
+  lifecycle {
+    ignore_changes = [
+      environment_properties,
+    ]
+  }
+}
```

Comment:
should this be removed?

Reply:
This cannot be removed. Because `environment_properties` is an O+C property.

Reply:
the schema does not have this marked as `Computed`?

Reply:
O+C literally means marked `Optional` and `Computed` in schema, I'm not sure I understand the requirement here? The use of `ignore_changes` in tests usually indicates an incorrect implementation in the provider that results in a value not being set (or unset) correctly, unless it's for a value that is expected to be modified by the use of another resource. Is that what we have here?

Reply:
I think this use case falls in this situation in the [doc](https://github.com/hashicorp/terraform-provider-azurerm/blob/660adeae29a5b32e795088b8181b36e2e045b9ce/contributing/topics/best-practices.md?plain=1#L226). There will be a value returned from the API if this field is not specified by the user, but it still can be configured if the user wants to.

Reply:
@catriona-m , changed as request.